query/call_iterator.go

```diff
@@ -39,6 +39,11 @@ func NewCallIterator(input Iterator, opt IteratorOptions) (Iterator, error) {
 	name := opt.Expr.(*influxql.Call).Name
 	switch name {
 	case "count":
+		// Count is optimized in the lower level storage to return an aggregate sum,
+		// per window, if there are no field conditions.
+		if opt.IndexOnlyConditions() {
```

Comment:
This shouldn't be here. If you have are merging multiple iterators into a call iterator, it will already use the sum iterator instead. So you just need to ensure that each series inside of the storage engine uses the optimized count cursor and the other logic should take care of the rest.

Look for the comment "Wrap each series in a call iterator". Since the decision to create the count cursor needs to be done inside of build cursor, it might be best to move the logic in this area into the creation of the iterators too.

Sorry if that's a bit confusing. Please tell me if you need help with that.